src/core/thread/mesh_forwarder.cpp

```diff
@@ -1479,9 +1517,9 @@ void MeshForwarder::HandleReassemblyTimer(void)
         }
     }
 
-    if (mReassemblyList.GetHead() != NULL)
+    if ((mReassemblyList.GetHead() != NULL) && (!mUpdateTimer.IsRunning()))
```

Comment:
Just curios, can we get here with `mUpdateTimer` running? 
We seem to call `UpdateReassemblyList()` from `HandleUpdateTimer()` where the timer should not be running.

Reply:
Changed.